src/Engine/GoogleCloudVisionEngine.php

```diff
@@ -43,6 +44,14 @@ public function getText(string $imageUrl, ?string $lang = null): string
             $this->gcv->setImage($imageUrl);
         } catch (LimitExceededException $e) {
             throw new OcrException('limit-exceeded', [$e->getMessage()]);
+        } catch (ErrorException $e) {
+            if (false !== strpos($e->getMessage(), '404 Not Found')) {
+                // The 'image-retrieval-failed' message is the important part that is localized.
+                throw new OcrException('image-retrieval-failed', ['404 Not Found']);
```

Comment:
Yeah, this was a bit of a hack trying to detect a simple 404 error. The GoogleCloudVision obscures any HTTP issue by running `file_get_contents` so the best we get is an `ErrorException` and plain text with the response code. I thought about maybe doing a HEAD request (since we don't need to fetch the image for Google) to the image in `EngineBase::checkImageUrl()`, that way we can get more meaningful response codes and return them in a consistent manner. How does that sound?

Reply:
I think with T277129 we'll be able to provide a URL directly, and not download the image file. In that case, when the URL doesn't exist we'll get back something like:

```
  -error: array:2 [▼
    "code" => 3
    "message" => "The URL does not appear to be accessible by us. Please double check or download the content and pass it in."
  ]
```

So maybe we just don't worry about it for now! :)

Reply:
Okay! Glad I asked :)